xbmc/URL.cpp

```diff
@@ -333,15 +333,15 @@ void CURL::SetFileName(const std::string& strFileName)
 {
   m_strFileName = strFileName;
 
-  int slash = m_strFileName.find_last_of(GetDirectorySeparator());
-  int period = m_strFileName.find_last_of('.');
-  if(period != -1 && (slash == -1 || period > slash))
+  size_t slash = m_strFileName.find_last_of(GetDirectorySeparator());
+  size_t period = m_strFileName.find_last_of('.');
+  if(period != string::npos && (slash == string::npos || period > slash))
     m_strFileType = m_strFileName.substr(period+1);
   else
     m_strFileType = "";
 
   slash = m_strFileName.find_first_of(GetDirectorySeparator());
-  if(slash == std::string::npos)
+  if(slash == string::npos)
```

Comment:
I don't approve of this, we want more std:: and less using namespace, this is going in the wrong direction